### PR TITLE
Wordpress Blocks Override Default Font Sizes with Figma Guided font sizes

### PR DESCRIPTION
**Pull request by MuluhGodson**
## Fixes

Fixes https://github.com/creativecommons/project_creativecommons.org/issues/110 by @brylie 

## Description
I used the Figma guides along with how the typography font sizes are defined in Vocabulary to work on this. 
In the `functions.php` file I used the font sizes given on Figma (which have `px` as units) given that the `functions.php` file accepts its values in `px`.
For the CSS rendered on the frontend, I used the `rem` units as is used in the Vocabulary package.

I didn't disable the possibility of manually entering a custom font size but if this is required I can quickly include it in another commit.
## Screenshots
image

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -47,6 +47,11 @@
 add_image_size( 'landscape-medium', 740, 416, true );
 add_image_size( 'landscape-featured', 1000, 500, true );
 
+//Media Upload size
+@ini_set( 'upload_max_size' , '64M' );
+@ini_set( 'post_max_size', '64M');
+@ini_set( 'max_execution_time', '300' );
+
```
Comment (brylie) [inline functions.php:54]: These configuration variables should typically be set at the project level rather than in a theme or plugin.

```suggestion
```

**Review by brylie — CHANGES_REQUESTED** [review]

**Comment by brylie** [comment]
Other than the PHP configuration variables, this PR looks good.

**Comment by MuluhGodson** [comment]
@brylie sorry for the php configuration variables. I don't remember touching them but I did a git stash in order to pull latest changes before committing. I think after applying the stash I may have accidentally touched them.

**Comment by brylie** [comment]
@MuluhGodson I'm sorry to flounder on this pull request a bit but I have some reservations about changing the font size semantics.

In particular, I'm concerned that using heading semantics might confuse content authors and reinforce the [common mistake of using headings to resize text](https://jackiedelia.com/dont-use-headings-to-make-text-larger/) rather than [describe the structure of a page](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements).

If anything, we should use the default Gutenberg font-size adjectives and adjust the sizes to track better with Vocabulary. For example, the Vocabulary base font size (16px?) would be used as the value for the Gutenberg "normal" font size with size increments of 1-2em for each step up/down the hierarchy.

image

It may be that the Gutenberg default font sizes are already close to the Vocabulary defaults, in which case we don't need to change anything. I'm always in favor of not writing code, as it means less overall maintenance costs and a reduced likelihood of bugs.

**Comment by MuluhGodson** [comment]
@brylie I agree. That was a concern too for me while writing them as I wasn't too sure about the semantics.

**Comment by brylie** [comment]
Let's put this task on hold while I clarify the related issue. I think there are some details I may have omitted.

**Comment by MuluhGodson** [comment]
I have readjusted as follows:
- The `H1` semantic has been changed to Wordpresses `Huge` as it's the closest font size (with actual font size updated to fit vocabulary).
- `H2` has been changed to `Large`
- `H3` has been changed to `medium`
-  H4, H5 and H6 have been removed.
- `body-big` has been changed to normal
- `caption` has been changed to small

**Comment by MuluhGodson** [comment]
> Let's put this task on hold while I clarify the related issue. I think there are some details I may have omitted.

Okay @brylie

File: style.css [hunk 2]
```diff
@@ -29,3 +29,28 @@ This theme, like WordPress, is licensed under the GPL.
     /* add overflow rule */
     overflow: hidden;
 }
+/*override wp font size css*/
+.has-small-font-size {
+    font-size: 0.8rem;
+}
+.has-body-normal-font-size {
+	font-size: 1rem;
```
Comment (brylie) [inline style.css:37]: That we are using `1rem` here may indicate this pull request is unnecessary.

Comment (MuluhGodson) [reply style.css:37]: yes very correct. the normal size on Wordpress is the same as Vocabulary's body normal which is the base line font size. I will proceed to close this PR.

**Pull request closed without merging**